src/styles/withStyles.d.ts

```diff
@@ -1,5 +1,11 @@
 import * as React from 'react';
 import { Theme } from './createMuiTheme';
+import * as CSS from 'csstype';
+
+export interface CSSProperties extends CSS.Properties<number | string> {
+  // Allow pseudo selectors and media queries
+  [k: string]: CSS.Properties<number | string>[keyof CSS.Properties] | CSSProperties;
```

Comment:
>Isn't ```CSS.Properties<number | string>[keyof CSS.Properties]``` already covered by ```extends CSS.Properties<number | string>```?

It's necessary for the string index to be compatible with the non-`string` properties. If you remove that it will be a type error.

>Also doesn't that make this valid?
>```
>color: {
>  color: 'red';
>}
>```

Allowing an arbitrary string index for nested pseudo selectors and media queries unavoidably means that invalid properties can get through. No way around it.